ewatercycle/observation/grdc.py

```diff
@@ -113,7 +114,14 @@ def _grdc_read(grdc_station_path, start, end):
     # Select GRDC station data that matches the forecast results Date
     grdc_station_select = grdc_station_df.loc[start:end]
 
-    return metadata, grdc_station_select
+    # Fix missing values
+    grdc_station_fix = grdc_station_select.copy()
```

Comment:
Why do you make a copy? Each call to `_grdc_read()` will generate a new dataframe there is no need to have 2 copies.

Reply:
Before adding this copy, I got this warning from pandas:

```
miniconda3/envs/comparison/lib/python3.8/site-packages/pandas/core/series.py:4575: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return super().replace(
```

Looking at the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy, it seems that copy() is needed after the selection in `grdc_station_df.loc[start:end]`.

Reply:
Ok, no warning is better. 
We could also add `na_values=-999` to [read_csv()](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html) on line 100 then we dont need any copy or replace.

Reply:
Thank you for the suggestion. Using na_values in pd.read_csv(), convert -999.000 to `nan` with type `numpy.float64` that is not equal to `np.nan` anymore. Thus, the test will break. I don't know how to represent the pandas `nan` value in the test where  `{'streamflow': ('time', [123., 456.,  np.NaN(?)])}`.

Reply:
Hmm, a quick test works for when I use `na_values='-999',`. 

```
cat b.csv 
a,b
1,2
3,-999
```

```python
import pandas as pd
import numpy as np
z = pd.read_csv('b.csv',na_values='-999')
type(z['b'][1])
numpy.float64
np.isnan(z['b'][1])
True
```

I think it is because np.nan is a special case of a floating number. So you can just use np.nan in expected value.

When I add `na_values='-999',` to arguments read_csv on line 100 the tests/observation/test_grdc.py test passes.

Reply:
thanks, it is fixed now.